packages/cosmos/src/gas_price.rs

```diff
@@ -162,22 +159,15 @@ struct OsmosisGasPrice {
     last_triggered: Instant,
 }
 
-/// Loads current eip base fee from a v1beta1 lcd endpoint
+/// Loads current eip base fee from Osmosis txfees module
 pub async fn load_osmosis_gas_base_fee(
-    client: &reqwest::Client,
+    builder: CosmosBuilder,
 ) -> Result<f64, LoadOsmosisGasPriceError> {
-    #[derive(serde::Deserialize)]
-    struct BaseFee {
-        base_fee: String,
-    }
-    let BaseFee { base_fee } = client
-        .get("https://lcd.osmosis.zone/osmosis/txfees/v1beta1/cur_eip_base_fee")
-        .send()
-        .await?
-        .error_for_status()?
-        .json()
-        .await?;
-    let base_fee: f64 = base_fee.parse()?;
+    // this cosmos instance is only used to load the gas price
+    let cosmos = builder.build_lazy()?;
```

Comment:
This isn't the approach we should take. This will require recreating connections and losing all the existing information in the real `Cosmos` value about node health.